code/parse/sexp.cpp

```diff
@@ -3843,11 +3841,20 @@ int get_sexp()
 		}
 
 		// update links
-		if (count++) {
+		if (start == -1) {
+			start = node;
+		} else {
 			Assert(last != -1);
+			if (last == start && (start == Locked_sexp_false || start == Locked_sexp_true)) {
```

Comment:
My own preference would be to write a comment before these new blocks explaining why they're there, to avoid mysterious special cases. Something like "Locked_sexp_true/false are only meant to represent operator nodes with no arguments, i.e. (true) and (false). If they appear as 'bare' arguments to another operator, or have arguments of their own, the global sexp node structure can become corrupted; we hack around this by always wrapping them in their own list in these cases and warning the user."

Reply:
This seemed like a really good summary of both the problem and solution, so I turned it directly into a comment.